shared/src/main/scala/org/akkajs/shocon/SHocon.scala

```diff
@@ -109,8 +110,11 @@ package object shocon extends Extractors {
                 case _ =>
                   k -> v
               }
-          } ++ mergeable.fields.filterKeys(diff.contains)
-
+          } ++ mergeable.fields.view.filter(e => diff.contains(e._1))
```

Comment:
This was also not compiling for scala 2.12 only.
I think `scala-collection-compat` is not covering this case.